src/main/java/frc/robot/Robot.java

```diff
@@ -50,15 +48,20 @@ public void robotInit() {
         
         DataLoggerFactory.configureForMatch();
         this.logger = DataLoggerFactory.getLoggerFactory().createDataLogger("Robot Main Loop");
+        subsystemManager = new SubsystemManager();
         subsystemManager.initAll();
 
         optionChooser = new SmartDashboardPathChooser();
+
         oi = new OperatorInterface(subsystemManager);
-        commandFactor = new CommandGroupFactory((subsystemManager));
+        commandFactory = new CommandFactory(subsystemManager);
     }
 
     @Override
     public void robotPeriodic() {
+        //runs after everything else
+        subsystemManager.updatePoses();
+        CommandScheduler.getInstance().run();
```

Comment:
So are we going to use robotPeriodic now? Are we essentially eliminating the use of the customperiodic?

Reply:
yeah it reduced ducplicated code. I read the api docs last night, and robotperiodic runs after everything else, so it will work for us. i think. i ended up changing my mind when i realized we'd have to duplicate the code AGAIN for test periodic